test/boost/mutation_reader_test.cc

```diff
@@ -2361,6 +2362,93 @@ SEASTAR_THREAD_TEST_CASE(test_multishard_streaming_reader) {
     }).get();
 }
 
+SEASTAR_THREAD_TEST_CASE(test_perf_multishard_streaming_reader) {
```

Comment:
Is this the test which was used to produce the comparison in the cover letter? This is a good starting point, but we need a macro benchmark with a cluster of nodes. Maybe we can use the reproducer @Deexie is working on.

Reply:
No, the result in the commit log is a real dev cluster with two nodes.  Yes, @Deexie 's reproducer would be helpful.